indico/modules/events/abstracts/forms.py

```diff
@@ -113,3 +123,53 @@ class CreateEmailTemplateForm(EditEmailTemplateRuleForm):
         ('reject', _('Reject')),
         ('merge', _('Merge'))
     ], description=_("The template that will be used as a basis for this notification. You can customize it later."))
+
+
+class AbstractForm(IndicoForm):
+    title = StringField(_("Title"), [DataRequired()])
+    description = IndicoMarkdownField(_('Content'), [DataRequired()], editor=True, mathjax=True)
+    submitted_contrib_type = QuerySelectField(_("Type"), get_label='name', allow_blank=True,
+                                              blank_text=_("No type selected"))
+    person_links = AbstractPersonLinkListField(_("People"), [DataRequired()])
+    submission_comment = TextAreaField(_("Comments"))
+    attachments = HiddenField(_('Attachments'), widget=DropzoneWidget(style='thin'))
+
+    def __init__(self, *args, **kwargs):
+        self.abstract = kwargs.pop('abstract', None)
+        super(AbstractForm, self).__init__(*args, **kwargs)
+        self.submitted_contrib_type.query = self.event.contribution_types
+        if not self.submitted_contrib_type.query.count():
+            del self.submitted_contrib_type
+        self.submitted_for_tracks.query = Track.query.with_parent(self.event).order_by(Track.title)
+        if abstracts_settings.get(self.event, 'allow_attachments'):
+            tpl = get_template_module('forms/_dropzone_themes.html')
+            self.attachments.widget.options['previewTemplate'] = tpl.thin_preview_template()
```

Comment:
Modifying this is not a good idea. When we move to a proper `DropzoneField` (which I think we should, see some of my other comments regarding `request.files`) this would modiy the widget instance which is usually set on the Field class level, i.e. shared between instances. So here you modify it for all instances of the field, not just this one.